File: judge-api/src/main/kotlin/io/leonfoliveira/judge/api/emitter/dto/emmit/SubmissionEmmitDTO.kt
package io.leonfoliveira.judge.api.emitter.dto.emmit

import io.leonfoliveira.judge.core.domain.entity.Submission
import io.leonfoliveira.judge.core.domain.enumerate.Language
import java.time.LocalDateTime

data class SubmissionEmmitDTO(
    val id: Int,
    val problem: ProblemDTO,
    val member: MemberDTO,
    val language: Language,
    val status: Submission.Status,
    val createdAt: LocalDateTime,
) {
    data class ProblemDTO(
        val id: Int,
        val title: String,
    )

    data class MemberDTO(
        val id: Int,
        val name: String,
    )
}

fun Submission.toEmmitDTO(): SubmissionEmmitDTO {
    return SubmissionEmmitDTO(
        id = this.id,
        problem =
            SubmissionEmmitDTO.ProblemDTO(
                id = this.problem.id,
                title = this.problem.title,
            ),
        member =
            SubmissionEmmitDTO.MemberDTO(
                id = this.member.id,
                name = this.member.name,
            ),
        language = this.language,
        status = this.status,
        createdAt = this.createdAt,
    )
}

File: judge-web/src/core/listener/dto/emmit/SubmissionEmmitDTO.ts
import { Language } from "@/core/domain/enumerate/Language";
import { SubmissionStatus } from "@/core/domain/enumerate/SubmissionStatus";

export type SubmissionEmmitDTO = {
  id: number;
  problem: {
    id: number;
    title: string;
  };
  member: {
    id: number;
    name: string;
  };
  language: Language;
  status: SubmissionStatus;
  createdAt: string;
};
